src/libraries/System.Security.Cryptography/tests/X509Certificates/CertificateCreation/CertificateRequestApiTests.cs

```diff
@@ -233,14 +278,36 @@ public static void CtorValidation_PublicKey_X500DN()
                 X509SignatureGenerator generator = X509SignatureGenerator.CreateForECDsa(ecdsa);
                 publicKey = generator.PublicKey;
             }
+        }
 
-            AssertExtensions.Throws<ArgumentNullException>(
-                "hashAlgorithm",
-                () => new CertificateRequest(subjectName, publicKey, default(HashAlgorithmName)));
+        [Fact]
+        public static void PublicKeyCtor_HashAlgorithm_LateVerification()
+        {
+            X500DistinguishedName name = new X500DistinguishedName("CN=Test");
+
+            using (ECDsa ecdsa = ECDsa.Create(EccTestData.Secp384r1Data.KeyParameters))
+            {
+                X509SignatureGenerator gen = X509SignatureGenerator.CreateForECDsa(ecdsa);
+
+                CertificateRequest req = new CertificateRequest(name, gen.PublicKey, default);
+                Assert.Null(req.HashAlgorithm.Name);
+                DateTimeOffset now = DateTimeOffset.UtcNow;
+
+                Assert.Throws<InvalidOperationException>(
+                    () => req.Create(name, gen, now.AddMinutes(-1), now.AddMinutes(1), new byte[] { 1, 2, 3 }));
+            }
 
-            AssertExtensions.Throws<ArgumentException>(
-                "hashAlgorithm",
-                () => new CertificateRequest(subjectName, publicKey, new HashAlgorithmName("")));
+            using (RSA rsa = RSA.Create(TestData.RsaBigExponentParams))
```

Comment:
Every time we use RSA with big exponents the test gets disabled on Android. https://github.com/dotnet/runtime/issues/72906#issuecomment-1458338955

Do we need to use big exponent params here?

Reply:
No, just that the rest of certreq does (as a "make sure we don't hard-code the RSA Exponent value").  I've gone ahead and changed it to some other pre-baked key for this test.